src/armory/Armory.js

```diff
@@ -1,34 +1,153 @@
 //@flow strict
 
 import * as React from 'react';
-//import { Link } from 'react-router-dom';
+import { Link } from 'react-router-dom';
 import Navbar from '../components/Navbar.js';
 
+// Tracks points to component names.
+const componentPoints = [

```

Comment:
I don't like this way of doing it. What you have if you do it this way is:
An array of objects, where each object has a single field (whose name is different every time), where that field maps to some integer.

Here are some issues that could cause:
- It is difficult to get the value for a particular component
- There is nothing stopping us from having two objects that have the same name (we might have two {moddableLight: ?}, and they might map to different values)
- There is nothing protecting us from typos here if someone spells 'moddableLight' as 'modableLight' or 'ModdableLight' or 'moddable_light' for example.

Other alternatives:
https://flow.org/en/docs/types/objects/#toc-objects-as-maps
or a typed version of this:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Either way, we need a flow type for the names of these parts so we don't get it wrong by accident.

Reply:
I'll look into the alternatives, although I'm probably going to save this issue for when the API is implemented.